### PR TITLE
Convert patches to quilt

### PR DESCRIPTION
**Pull request by freeekanayaka**
Instead of keeping patches against Juju core code in a single patch
file, use quilt to manage a series of more granural patch files.

This makes it both easier to develop new patches and also to
selectively apply to certain target Juju versions.

It also makes it more clear what the purpose of a certain patch is, as
it can be documented in the patch file itself. I indeed added documentation
for the quilt patch files in this branch that I extracted from the existing
single-file patch.

Basic documentation about how to use quilt for fake-juju is
being added in a follow-up branch.

See:

https://github.com/juju/fake-juju/blob/more-devel-docs/HACKING.rst#adding-a-new-patch

for a preview.

### CODE REVIEW

File: common/patches/dont-crash-if-no-test-charm-repo-is-there.diff [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+===================================================================
+--- 2.0.2.orig/src/github.com/juju/juju/testcharms/charm.go
++++ 2.0.2/src/github.com/juju/juju/testcharms/charm.go
+@@ -17,7 +17,9 @@ import (
+ )
+ 
+ // Repo provides access to the test charm repository.
++// XXX fake-juju: start patch
+ var Repo = testing.NewRepo("charm-repo", "quantal")
++///XXX fake-juju: end patch
```
Comment (sparkiegeek) [inline common/patches/dont-crash-if-no-test-charm-repo-is-there.diff]: Am I missing something here? This appears to be a no-op patch?

File: common/patches/fake-cli-bootstrap.diff [hunk 2]
```diff
@@ -0,0 +1,20 @@
+Modify the implementation of the "juju bootstrap" command line so
+instead of going through the regular logic it calls a the
```
Comment (sparkiegeek) [inline common/patches/fake-cli-bootstrap.diff]: s/a the/the/

File: common/patches/dont-crash-if-no-test-charm-repo-is-there.diff (hunk 1 above)
Comment (freeekanayaka) [inline common/patches/dont-crash-if-no-test-charm-repo-is-there.diff]: Ugh! Good catch, fixed it to be:

```
-var Repo = testing.NewRepo("charm-repo", "quantal")
+// XXX fake-juju: start patch
+var Repo = &testing.Repo{}
+// XXX fake-juju: end patch
```

like it is currently in juju-core.patch.

File: common/patches/fake-cli-bootstrap.diff (hunk 2 above)
Comment (freeekanayaka) [inline common/patches/fake-cli-bootstrap.diff]: Fixed

**Pull request merged**